### DISCUSSION TITLE
🙏How to freeze non-code/assets files into firmware and access later & how to modify default filesystem contents.🙏

### DISCUSSION THREAD

**Discussion by sethstenzel** [micropython/micropython#10944] [ESP32] [answered]
I'm not sure about the rules for reposting or getting eyes on an [old post](https://github.com/orgs/micropython/discussions/10707), so I'm just going to ask the same question again. This is a pretty important question to me, and I'm sure other newbies would benefit from a solid answer(s). I'd also be willing to blog a complete answer if someone just wants to vomit all the links/documentation/forum posts I need to piece the information together myself if they don't have time for the hand-holding.

I searched the old forums, but the information is either not there, or I don't know the right keywords/terms to access it.
I'd be more than willing to pay / tip via Paypal or something for the hand-holding and information that actually helps me achieve my goal if that incentivizes an answer / makes it at least a little worths someone's time to answer. 
_(I'm not trying to cheapen or besmirch the community that is obviously very supportive with his comment, I just really want this answered.)_

I'm looking for help, or a pointer to some documentation on how to "freeze" non-python files in firmware such as images, sounds, or other compressed assets into the firmware.

I currently have a built toolchain setup in WSL, and can build custom images for my board. I can create .mpy, or add .py files to be frozen into firmware but I don't see documentation on how to freeze other things like graphics, or sounds into firmware, or how those assets would be accessed. I also can't seem to find how I would modify the default file system with additional files, like settings files along side the current defaults of main.py and boot.py.

Is there a way to add additional default files to the file system, like boot.py and main.py are added automatically now?

I'm dreading the idea that I might have to convert all my assets to some sort of .py file containing binary data.

Has anyone done this, knows how to do this, or can you point me to documentation specifically on these questions?

Thanks all!

**Comment by Josverl** [comment]
It would have been better IMO to reply with a nudge to your prior post.

There may be an answer waiting there....

**Reply by sethstenzel** [reply]
Hey, thanks for your reply, I had no idea if commenting on my previous post would push it to the top of the stack, and I don't see anything on the page called a "nudge" or anything like that or I would have.

**Comment by sethstenzel** [comment, marked as answer]
Answered here: [https://github.com/orgs/micropython/discussions/10707#discussioncomment-5221425](https://github.com/orgs/micropython/discussions/10707#discussioncomment-5221425)
